### ISSUE TITLE
Default Value of fail_if_no_peer_cert

### ISSUE THREAD

**Issue by tigercl** [emqx/emqx#13253]
### What happened?

Now when I enable "Verify Peer" for SSL Listener, the default value of "Force Verify Peer Certificate"(`fail_if_no_peer_cert`) will be **false**.

![image](https://github.com/emqx/emqx/assets/15085956/a3c79bcf-c756-48d7-8902-e64649d162c0)

Since this is not a common option, users may not change it. But it may have some security risks: when set to false, even if the user has enabled two-way authentication, the client can still connect successfully without providing a certificate.

This means that attackers may be able to easily access the user's server, because when enabling two-way authentication, users usually no longer enable password-based authentication.

### What did you expect to happen?

I hope EMQX can change the default value of `fail_if_no_peer_cert` to True, so that users can explicitly change it to False when needed.

### EMQX version

EMQX 5.7

### OS version

macOS 13

### Log files

<details>

</details>

**Comment by zmstone** [comment]
@ysfscream this is more of a frontend enhancement?

**Comment by tigercl** [comment]
@zmstone I don't think so, the default value of this option in the configuration file is also false.

**Comment by zmstone** [comment]
> Why do we even have this option? Is there a scenario when a user may want to allow plain tcp connection on SSL port?

Good point.

**Comment by zmstone** [comment]
> @zmstone I don't think so, the default value of this option in the configuration file is also false.

We can change default value from backend, but frontend is independent implementation.

**Comment by zmstone** [comment]
Here is my proposal.
In 5.8, we will only expose "verify peer" option in config and in dashboard, always force client to send certificate when "verify_peer".
For backward compatibility, we will hide `fail_if_no_peer_cert` from backend schema.

**Comment by tigercl** [comment]
@id When `fail_if_no_peer_cert` is false, the client still establishes a TLS connection to the server, it just won't be asked for a client certificate.

The only scenario I can think of is that all the user's clients access from the same TLS port, some of them provide client certificates for two-way authentication, and these clients will skip the password authentication of EMQX. The other part does not provide client certificates, and they must pass password authentication to connect.

However, I think only HTTP authentication currently supports this process.

**Comment by tigercl** [comment]
> Here is my proposal. In 5.8, we will only expose "verify peer" option in config and in dashboard, always force client to send certificate when "verify_peer". For backward compatibility, we will hide `fail_if_no_peer_cert` from backend schema.

emmm What happens to listeners that had `fail_if_no_peer_cert` set to false?

**Comment by zmstone** [comment]
> > Here is my proposal. In 5.8, we will only expose "verify peer" option in config and in dashboard, always force client to send certificate when "verify_peer". For backward compatibility, we will hide `fail_if_no_peer_cert` from backend schema.
> 
> emmm What happens to listeners that had `fail_if_no_peer_cert` set to false?

it will be discarded

**Comment by qzhuyan** [comment]
> Why do we even have this option? Is there a scenario when a user may want to allow plain tcp connection on SSL port?

Peer sending its cert is sometimes optional for TLS client (not limited to MQTT).
